source/common/common/utility.cc

```diff
@@ -107,3 +107,16 @@ const std::string& StringUtil::valueOrDefault(const std::string& input,
                                               const std::string& default_value) {
   return input.empty() ? default_value : input;
 }
+
+std::string StringUtil::replaceAll(const std::string& input, const std::string& search,
+                                   const std::string& replacement) {
+  std::string input_ = input;
```

Comment:
minor: x_ is naming convention for class fields. result or something like that would work here.